File: Tutoriels/Python/POO/src/manager/GProcessHello.py
#================================================
from GProcess import GProcess
#================================================
class GProcessHello(GProcess):
    #================================================
    __instance = None
    #================================================
    def __init__(self):
        if GProcessHello.__instance == None:
            GProcessHello.__instance = self        
    #================================================
    @staticmethod
    def Instance():
        if GProcessHello.__instance == None:
            GProcessHello()
        return GProcessHello.__instance
    #================================================
    def run(self):
        print "Hello Sub"
#================================================

File: Tutoriels/CSharp/POO/src/manager/GProcessHello.cs
//===============================================
using System;
using System.Collections.Generic;
//===============================================
public class GProcessHello : GProcess {
    //===============================================
    private static GProcessHello m_instance;
    static readonly object m_lock = new object();
    //===============================================
    private GProcessHello() {
		
    }
    //===============================================
    public static GProcessHello Instance() {
        lock(m_lock) {
            if(m_instance == null) {
               m_instance = new GProcessHello();
            }
	   	    return m_instance;
        }
    }    
    //===============================================
    public override void run(string[] args) {
        GPrint.Instance().print("Bonjour tout le monde");
    }
    //===============================================
}
//===============================================
